app/src/main/java/com/infomaniak/mail/views/AvatarView.kt

```diff
@@ -196,15 +196,17 @@ class AvatarView @JvmOverloads constructor(
     ) {
         when (avatarDisplayType) {
             AvatarDisplayType.UNKNOWN_CORRESPONDENT -> loadUnknownUserAvatar()
+            AvatarDisplayType.USER_AVATAR -> loadUserAvatar(correspondent!!, bimi)
             AvatarDisplayType.CUSTOM_AVATAR,
             AvatarDisplayType.INITIALS -> loadAvatarUsingDictionary(correspondent!!, contacts, bimi)
-            AvatarDisplayType.BIMI -> loadBimiAvatar(bimi!!, correspondent!!)
+            AvatarDisplayType.BIMI -> loadBimiAvatar(correspondent!!, bimi!!)
         }
     }
 
     private fun getAvatarDisplayType(correspondent: Correspondent?, bimi: Bimi?, isBimiEnabled: Boolean): AvatarDisplayType {
         return when {
             correspondent == null -> AvatarDisplayType.UNKNOWN_CORRESPONDENT
+            correspondent.isMe() -> AvatarDisplayType.USER_AVATAR
```

Comment:
Why not use `shouldDisplayUserAvatar()` instead? The method feels better suited for what you're trying to do